internal/bminventory/inventory.go

```diff
@@ -996,6 +997,86 @@ func (b *bareMetalInventory) updateImageInfoPostUpload(ctx context.Context, infr
 	return nil
 }
 
+func (b *bareMetalInventory) updateExternalImageInfo(infraEnv *common.InfraEnv, infraEnvProxyHash string, imageType models.ImageType) error {
+	updates := map[string]interface{}{}
+
+	// this is updated before now for the v2 (infraEnv) case, but not in the cluster ISO case so we need to check if we should save it here
+	if infraEnv.ProxyHash != infraEnvProxyHash {
+		updates["proxy_hash"] = infraEnvProxyHash
+		infraEnv.ProxyHash = infraEnvProxyHash
+	}
+
+	var (
+		prevType    string
+		prevVersion string
+		prevArch    string
+	)
+	if infraEnv.DownloadURL != "" {
+		currentURL, err := url.Parse(infraEnv.DownloadURL)
+		if err != nil {
+			return errors.Wrap(err, "failed to parse current download URL")
+		}
+		vals := currentURL.Query()
+		prevType = vals.Get("type")
+		prevVersion = vals.Get("version")
+		prevArch = vals.Get("arch")
+	}
+
+	updates["type"] = imageType
+	infraEnv.Type = imageType
+
+	osImage, err := b.getOsImageOrLatest(&infraEnv.OpenshiftVersion, infraEnv.CPUArchitecture)
+	if err != nil {
+		return err
+	}
+
+	var version string
+	if osImage.Version != nil {
+		version = *osImage.OpenshiftVersion
+	} else {
+		return errors.Errorf("OS image entry '%+v' missing OpenshiftVersion field", osImage)
+	}
+
+	var arch string
+	if osImage.CPUArchitecture != nil {
+		arch = *osImage.CPUArchitecture
+	}
+
+	// only update download url if type or version have changed
```

Comment:
I think this actually isn't the behavior we want.

I did a test that updated the ignition config, which didn't cause the url to be updated.
That's actually bad because we would want to trigger a new image download in the ztp case for an ignition change.

Is forcing a new api key the only way we can bounce the host in this scenario? @flaper87 ?

Reply:
I think the API Key bump is the easiest since that will make the image URL different, which is the only thing we use right now to know whether there's a new image.

We talked a bit about this in slack and here are 2 considerations: 

1. I would try to think whether we really care about supporting v1 or not. I don't think the image service will (or should for that matter) be used with v1
2. When designing the "re-deploy" logic, let's keep in mind the [agent reboot enhancement](https://github.com/openshift/assisted-service/blob/master/docs/enhancements/bmac-reboot-hosts-on-iso-change.md#proposal) which should (eventually) dictate how this whole logic works.

hope this helps